app/models/user.rb

```diff
@@ -48,6 +48,10 @@ class User < ApplicationRecord
   has_many :throttles, dependent: :destroy
   has_one :registration_log, dependent: :destroy
   has_one :proofing_component, dependent: :destroy
+  has_many :service_providers,
+           -> { merge(Identity.not_deleted) },
```

Comment:
1. Is this a confusing condition to put on here? I could also move this scope into the call site
2. Is it OK to assume that we **do not** want to notify revoked SPs of credential changes?

Reply:
I think if it's the only use of that scope, moving to call site makes sense.

My expectation as a user would be no more communication once it's been revoked. In what cases would an SP need to know a disconnected user has changed emails?

Reply:
1. ok! moved in 72c754cac2c6753ad346a5f0b27c9b134b53d7a1

2. So in the case where a user disconnects, changes emails, then reconnects, the SP might not have the right information? Like if the SP is expecting the old email? Some poorly behaved SPs have done this. But I guess, poorly-behaved SPs might not even subscribe to this event anyways, probably a moot point

Reply:
Is using UUID the recommended way to maintain that connection?

Reply:
Yeah UUID is recommended